src/ucp/api/ucp.h

```diff
@@ -2100,6 +2123,33 @@ ucs_status_t ucp_context_query(ucp_context_h context_p,
                                ucp_context_attr_t *attr);
 
 
+/**
+ * @ingroup UCP_MEM
+ * @brief Compare two remote keys
+ *
+ * This routine returns equal if all the transports can compare equal and
+ * configurations are equal.
+ *
+ * The comparison result is < 0 if rkey1 is lower than rkey2, 0 if they
+ * are equal or > 0 if rkey1 is greater than rkey2. The remote key greater-than
+ * semantic is arbitrary and implementation dependent, nevertheless it's
+ * guaranteed to be consistent (transitive and antisymmetric). It can be used,
+ * for example, to perform an efficient binary search in a sorted array of
+ * remote keys.
+ *
+ * @param [in]  context     Handle to @ref ucp_context_h
+ * @param [in]  rkey1       First rkey to compare
+ * @param [in]  rkey2       Second rkey to compare
+ * @param [in]  params      Additional parameters to the comparison
+ * @param [out] result      Result of the comparison
+ *
+ * @return Error code or UCS_OK if result contains the comparison result
```

Comment:
In what cases do we return an error ? Can one detect rkey corruption ?

Reply:
Not really sure what error will mean here...

Reply:
added list of error cases